### PR TITLE
Kernel source changes

### PR DESCRIPTION
**Pull request by Whissi**
WIP. Early draft. Motivation/purpose will follow.

### CODE REVIEW

**Comment by Whissi** [comment]
@gentoo/kernel @floppym

**Comment by kuzetsa** [comment]
re: virtual for LTS branch pinning

migrating to use branch & subslots would have the same effect:

for example:

```
emerge sys-kernel/gentoo-sources:4.14
```

---

adding USE build-deps should resolve an issue which repoman has been nagging about :+1:

**Comment by kuzetsa** [comment]
> > re: virtual for LTS branch pinning
> > migrating to use branch & subslots 
[...]
> 
> No: Kernel sources have set whole $PV as sub slot. That's why we need a "new" package for this. Of course, we could discuss to change this...

right. I didn't mean it was currently set up that way, hence the word: "*migrating*"

sorry for the confusion

File: virtual/linux-sources-build-deps/linux-sources-build-deps-1.ebuild [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+DESCRIPTION="Virtual package for linux sources build dependencies"
```
Comment (Zlogene) [inline virtual/linux-sources-build-deps/linux-sources-build-deps-1.ebuild]: it is not virtual by defenition, it does have providers, so this probably should be moved to sys-kernel/

Comment (aliceinwire) [reply virtual/linux-sources-build-deps/linux-sources-build-deps-1.ebuild]: moving to sys-kernel/ would made virtual/gentoo-sources collide with sys-kernel/gentoo-sources

Comment (Whissi) [reply virtual/linux-sources-build-deps/linux-sources-build-deps-1.ebuild]: No, @Zlogene commented on the *virtual/linux-sources-build-deps* not *virtual/gentoo-sources*. *virtual/linux-sources-build-deps* could become *sys-kernel/linux-sources-build-deps* from this POV.

Comment (aliceinwire) [reply virtual/linux-sources-build-deps/linux-sources-build-deps-1.ebuild]: ok yes.

**Comment by mjeveritt** [comment]
Although we might not be ready to merge this, is it worth refreshing @Whissi ?

**Comment by SoapGentoo** [comment]
over 3.5 years old

**Pull request closed without merging**